x-pack/plugins/alerting/server/rules_client/tests/bulk_enable.test.ts

```diff
@@ -0,0 +1,624 @@
+            version: '1',
+          } as SavedObject,
+        ],
+      });
+      taskManager.bulkEnable.mockImplementation(
+        async () =>
+          ({
+            tasks: [{ id: 'taskId1' }, { id: 'taskId2' }],
+            errors: [
+              {
+                task: {
+                  id: 'taskId2',
+                },
+                error: {
+                  error: '',
+                  message: 'UPS',
+                  statusCode: 500,
+                },
+              },
+            ],
+          } as unknown as BulkUpdateTaskResult)
+      );
+
+      const result = await rulesClient.bulkEnableRules({ filter: 'fake_filter' });
+
+      expect(logger.debug).toBeCalledTimes(1);
+      expect(logger.debug).toBeCalledWith(
+        'Successfully enabled schedules for underlying tasks: taskId1'
+      );
+      expect(result).toStrictEqual({
+        errors: [],
+        total: 2,
+        taskIdsFailedToBeEnabled: ['taskId2'],
+      });
+    });
+
+    test('should not throw an error if taskManager throw an error', async () => {
+      unsecuredSavedObjectsClient.bulkCreate.mockResolvedValue({
+        saved_objects: [
+          {
+            id: 'id1',
+            version: '1',
+          } as SavedObject,
+          {
+            id: 'id2',
+            version: '1',
+          } as SavedObject,
+        ],
+      });
+      taskManager.bulkEnable.mockImplementation(() => {
+        throw new Error('UPS');
+      });
+
+      const result = await rulesClient.bulkEnableRules({ filter: 'fake_filter' });
+
+      expect(logger.error).toBeCalledTimes(1);
+      expect(logger.error).toBeCalledWith(
+        'Failure to enable schedules for underlying tasks: taskId1, taskId2. TaskManager bulkEnable failed with Error: UPS'
+      );
+      expect(result).toStrictEqual({
+        errors: [],
+        taskIdsFailedToBeEnabled: ['taskId1', 'taskId2'],
+        total: 2,
+      });
+    });
+  });
+
+  describe('auditLogger', () => {
+    jest.spyOn(auditLogger, 'log').mockImplementation();
+
+    test('logs audit event when enabling rules', async () => {
+      unsecuredSavedObjectsClient.bulkCreate.mockResolvedValue({
+        saved_objects: [
+          {
+            id: 'id1',
+            version: '1',
+          } as SavedObject,
+        ],
+      });
+
+      await rulesClient.bulkEnableRules({ filter: 'fake_filter' });
+
+      expect(auditLogger.log.mock.calls[0][0]?.event?.action).toEqual('rule_enable');
+      expect(auditLogger.log.mock.calls[0][0]?.event?.outcome).toEqual('unknown');
+      expect(auditLogger.log.mock.calls[0][0]?.kibana).toEqual({
+        saved_object: { id: 'id1', type: 'alert' },
+      });
+      expect(auditLogger.log.mock.calls[1][0]?.event?.action).toEqual('rule_enable');
+      expect(auditLogger.log.mock.calls[1][0]?.event?.outcome).toEqual('unknown');
+      expect(auditLogger.log.mock.calls[1][0]?.kibana).toEqual({
+        saved_object: { id: 'id2', type: 'alert' },
+      });
```

Comment:
We're using this test helper function to simplify audit logging related assertions. Might be useful for your tests as well.

```
function expectAuditEvents(...events: ExpectedAuditEvent[]) {
  expect(auditLogger.log).toHaveBeenCalledTimes(events.length);
  for (let i = 0; i < events.length; i++) {
    const { action, outcome, kibana } = events[i];
    expect(auditLogger.log).toHaveBeenNthCalledWith(
      i + 1,
      expect.objectContaining({
        event: { action, category: ['authentication'], ...(outcome && { outcome }) },
        ...(kibana && { kibana }),
      })
    );
  }
}
```

You can use it like so:

```
expectAuditEvents(
  { action: 'user_login', outcome: 'success' },
  { action: 'user_logout', outcome: 'unknown' },
);
```